src/Components/Lifecycle/Lifecycle.tsx

```diff
@@ -136,35 +141,36 @@ const LifecycleTab: React.FC<React.PropsWithChildren> = () => {
     let dataToProcess: Stream[] | SystemLifecycleChanges[] = [];
     let hasInstalledData = false;
 
-    if (value === DEFAULT_DROPDOWN_VALUE || value === RHEL_8_STREAMS_DROPDOWN_VALUE) {
+    if ([DEFAULT_DROPDOWN_VALUE, RHEL_8_STREAMS_DROPDOWN_VALUE, RHEL_10_STREAMS_DROPDOWN_VALUE].includes(value)) {
       dataToProcess = filterAppDataByDropdown(fullAppLifecycleChanges, value);
       setAppLifecycleChanges(dataToProcess);
 
-      // Check if we have installed app data for the selected dropdown
-      const installedAppStreams = filterAppDataByDropdown(installedAppData, value);
+      // Check if we have installed and related app data for the selected dropdown
+      const installedAppStreams = filterAppDataByDropdown(relatedAppData, value);
       hasInstalledData = installedAppStreams.length > 0;
     } else if (value === RHEL_SYSTEMS_DROPDOWN_VALUE) {
       dataToProcess = systemLifecycleChanges;
 
       // Check if we have installed system data
-      hasInstalledData = installedSystemData.length > 0;
+      hasInstalledData = relatedSystemData.length > 0;
     } else {
       // Fallback for any unexpected dropdown value - use app streams as default
       dataToProcess = filterAppDataByDropdown(fullAppLifecycleChanges, DEFAULT_DROPDOWN_VALUE);
 
       // Check installed app data for default dropdown
-      const installedAppStreams = filterAppDataByDropdown(installedAppData, DEFAULT_DROPDOWN_VALUE);
+      const installedAppStreams = filterAppDataByDropdown(relatedAppData, DEFAULT_DROPDOWN_VALUE);
       hasInstalledData = installedAppStreams.length > 0;
     }
 
-    // Always set noDataAvailable based on whether there's installed data
-    // This controls whether the installed-only and installed-and-related views are disabled
-    setNoDataAvailable(!hasInstalledData);
+    // Only set noDataAvailable when relatedAppData is empty
+    setNoDataAvailable(relatedAppData.length === 0);
```

Comment:
With this change we don't need the `hasInstalledData` as we are only setting it in the if else chain above and not checking it's value anywhere. I would prefer using the original behavior which was:
```suggestion
    setNoDataAvailable(!hasInstalledData);
```